compiler/rustc_errors/src/lib.rs

```diff
@@ -1754,7 +1754,7 @@ impl DelayedDiagnostic {
             BacktraceStatus::Captured => {
                 let inner = &self.inner;
                 self.inner.subdiagnostic(DelayedAtWithNewline {
-                    span: inner.span.primary_span().unwrap(),
+                    span: inner.span.primary_span().unwrap_or(DUMMY_SP),
```

Comment:
This is a drive-by: We were ICEing here when reporting a delayed good-path bug since those don't have spans.